src/Resolvers/Microsoft.NET.Sdk.WorkloadResolver/MSBuildWorkloadSdkResolver.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+
+#if NET
+using Microsoft.DotNet.Cli;
+#else
+using Microsoft.DotNet.DotNetSdkResolver;
+#endif
+
+#nullable disable
+
+namespace Microsoft.NET.Sdk.WorkloadResolver
+{
+    public class MSBuildWorkloadSdkResolver : SdkResolver
+    {
+        public override string Name => "Microsoft.DotNet.MSBuildWorkloadSdkResolver";
+
+        public override int Priority => 4000;
+
+        private IWorkloadManifestProvider _workloadManifestProvider;
+        private IWorkloadResolver _workloadResolver;
+
+
+#if NETFRAMEWORK
+        private readonly NETCoreSdkResolver _sdkResolver;
+#endif
+
+        public MSBuildWorkloadSdkResolver()
+        {
+#if NETFRAMEWORK
+            _sdkResolver = new NETCoreSdkResolver();
+#endif
+        }
+
+        private void InitializeWorkloadResolver(SdkResolverContext context)
+        {
+            var dotnetRootPath = GetDotNetRoot(context);
+
+            var sdkDirectory = GetSdkDirectory(context);
+            //  TODO: Is the directory name OK, or should we read the .version file (we don't want to use the Cli.Utils library to read it on .NET Framework)
+            var sdkVersion = Path.GetFileName(sdkDirectory);
+
+
+            _workloadManifestProvider ??= new SdkDirectoryWorkloadManifestProvider(dotnetRootPath, sdkVersion);
+            //  TODO: Fix namespace / type names so there's not a collision for WorkloadResolver
+            _workloadResolver ??= new Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver(_workloadManifestProvider, dotnetRootPath);
+        }
+
+        public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext resolverContext, SdkResultFactory factory)
+        {
+            InitializeWorkloadResolver(resolverContext);
+
+            if (sdkReference.Name.Equals("Microsoft.NET.SDK.WorkloadAutoImportPropsLocator", StringComparison.OrdinalIgnoreCase))
+            {
+                List<string> autoImportSdkPaths = new List<string>();
+                foreach (var sdkPackInfo in _workloadResolver.GetInstalledWorkloadPacksOfKind(WorkloadPackKind.Sdk))
+                {
+                    string sdkPackSdkFolder = Path.Combine(sdkPackInfo.Path, "Sdk");
+                    string autoImportPath = Path.Combine(sdkPackSdkFolder, "AutoImport.props");
+                    if (File.Exists(autoImportPath))
+                    {
+                        autoImportSdkPaths.Add(sdkPackSdkFolder);
+                    }
+                }
+                return factory.IndicateSuccess(autoImportSdkPaths, sdkReference.Version);
+            }
+            else if (sdkReference.Name == "TestSdk")
```

Comment:
This is temporary? Add a hook for test, and we should do the rest in the test setup

Reply:
Oops, yes, I've now removed this.